src/app/query/ProductList.query.js

```diff
@@ -167,8 +174,9 @@ export class ProductListQuery {
     }
 
     _getProductField() {
+        const { getLinksFieldFromCart } = this.options;
         return new Field('product')
-            .addFieldList(this._getProductInterfaceFields(true));
+            .addFieldList(this._getProductInterfaceFields(true, getLinksFieldFromCart));
```

Comment:
I do not like this logic it is too complicated! Why is this even related to query? Can you determine if you want this product links or not in reducer?